guides/v2.2/contributor-guide/contributing_dod.md

```diff
@@ -57,74 +57,115 @@ Any backward-incompatible changes must also be recorded in the accompanying docu
+
+### Integration Tests
+
+An integration test should be used to cover any code that doesn't meet the requirements of a unit test or functional test and should be thought of as essentially the default test type.
+
+Integration tests come in many forms and can drastically vary in definition depending on what the intention of the test is. Generally speaking an integration test verifies that two or more things work together correctly. i.e. We are testing the _integration_ of one thing with another thing. 
+For the purpose of Magento testing, there are essentially two broad categories of integration tests: narrow-form and broad-form.
+
+* Narrow-form integration tests focus on testing something mostly in isolation an can be as simple as a unit test as described in this guide except without mocked dependencies.
+* Broad-form integration tests focus on testing something with less isolation and will involve multiple components explicitly collaborating. (compared to narrow-form where the dependencies are implicit collaborating)
+
+Integration test policy:
+
+* Methods and classes must have blackbox or greybox test coverage making sure to include all variations for normal use as well as corner-cases. 
+* SPI's must have test coverage. Often times Magento has extension points that may only be utilized by extension developers. Use TestModule's to implement those SPI extension points and verify they are used correctly. 
+* As mentioned above, classes marked with `@api` must contain coverage. Integration tests must be used cover these classes unless a unit test is more appropriate.
+* Consumers of default SPI's implementations should have at least basic coverage that ensures the default implementations of the SPI are correctly configured and loaded.  
+  
+  For example, `Magento\Framework\SomeClassFilter` may contain a `FilterPool` that comes with default `FilterInterface`'s from `Magento\Framework`. Each of these implementations would have their own coverage pursuant to this document. However, there should also be some basic assertions within the test coverage for `Magento\Framework\SomeClassFilter` that ensure each of the default filters are loaded correctly.     
+  This shouldn't be explicitly coverage such as `$filter->isLoaded('someDefaultFilter')` 
+
+See: [Running Integration Tests][3].
 
 ### Functional Tests
 
-Must cover new or changed application behavior (functional).
-Added/updated functionality should be covered by a functional autotest(s) related to previous sprint commitment in scope of current sprint.
+UI functional tests are inherently unstable irrespective of platform or testing framework. For this reason along with reasons of maintainability and quality assurance delivery time, these tests should be prioritized below all other tests types.
```

Comment:
I think it is reasonable to use functional tests to cover user behavior scenarios. When we are talking about the user story, it makes sense to cover S0, S1 scenario with UI tests. About maintainability, it can be improved by refactoring test-related entity, removing code duplication.

Reply:
Yes, it is still required to cover the P0/P1 scenarios with functional test coverage. UI test coverage in the case of user behavior. These scenarios have to be prioritized and defined by the product owner who should have knowledge of which behavioral flows are important enough to deserve dedicated system-level blackbox tests.

Reply:
> Even if tests are unreliable, it should be still more efficient to run them during regression than doing manual testing.

More efficient in what way though? Not seeing a real failed test right away and waiting for code to go through a full PR process then waiting for a pagerduty team to pick it up and investigate the issue, understand what happened, try and understand who broke it, then either fix the test or try to fix the behavior, seems like a way less efficient process then just having a reliable test suite.


> To optimize every day work, functional tests can be divided into groups (which I thought we already do) and only selected scenarios run on every PR.

For functional tests, yes this would be desirable and there is a separate initiative that hasn't been started to selectively run functional tests based on what was actually changed.